components/learning/PublicContent.vue

```diff
@@ -10,14 +10,16 @@
     <div class="public-course__units-list">
       <div
         v-for="unit of course.units"
-        :key="unit"
+        :key="unit.id"
         class="public-course__units-item"
       >
         <ProgressCircle :progress="0" />
         <BasicLink
-          :to="`/course/${course.slug}/${unitStore.unitById(unit).value?.slug}`"
+          :to="`/course/${course.slug}/${
+            unitStore.unitById(unit.id).value?.slug
+          }`"
         >
-          {{ unitStore.unitById(unit).value?.title }}
+          {{ unitStore.unitById(unit.id).value?.title }}
```

Comment:
```suggestion
          {{ unitStore.unitById(unit.id).value!.title }}
```

I think we should assume that the unit exits (with the `!` instead of `?`).

And if not, we should control what to render if there is no unit.
But I believe that it cannot be the case that there is no unit here.

Reply:
Sorry for the confusion, but this component is not part of this refactor. I only modified the `unit` to be `unit.id`, because I changed the returned value on the service and this page broke

Reply:
Ok!